### ISSUE TITLE
[Bug]: Unexpected behavior of `returned_token_ids` in Reward Modeling for LlamaForCausalLM

### ISSUE THREAD

**Issue by ryokamoi** [vllm-project/vllm#16545]
### Your current environment

<details>
<summary>The output of `python collect_env.py`</summary>

```text
INFO 04-12 22:41:00 [__init__.py:239] Automatically detected platform cuda.
Collecting environment information...
PyTorch version: 2.6.0+cu124
Is debug build: False
CUDA used to build PyTorch: 12.4
ROCM used to build PyTorch: N/A

vLLM Version: 0.8.2
vLLM Build Flags:
```

</details>

### 🐛 Describe the bug

The [reward modeling documentation](https://docs.vllm.ai/en/latest/models/supported_models.html#reward-modeling-task-reward) provides the following example for using `LlamaForCausalLM` with `--task reward`:

> For process-supervised reward models such as peiyi9979/math-shepherd-mistral-7b-prm, the pooling config should be set explicitly, e.g.: --override-pooler-config '{"pooling_type": "STEP", "step_tag_id": 123, "returned_token_ids": [456, 789]}'.

However, the usage of `returned_token_ids` leads to unexpected behavior.

Specifically, `returned_token_ids` is used to index into `hidden_states`, which are the intermediate representations before the final layer (4096 dimensions in Llama), **not the logits**. Using vocabulary token IDs `returned_token_ids` to select from these intermediate feature vectors doesn’t make sense.

https://github.com/vllm-project/vllm/blob/6c11ecf8d337d3b89e891588c81640a0bd30f6e1/vllm/model_executor/layers/pooler.py#L192-L193

For example, setting a large number (e.g., greater than 4096) in `returned_token_ids` results in an error.

When using `LlamaForCausalLM` for reward modeling with the current version of the code, intermediate representations should be extracted without setting `returned_token_ids` and then projected to logits afterward.

I also note that `step_tag_id` should be set to the token **before** the target position because `LlamaForCausalLM` is a causal model.

### Suggestions
* It would be helpful to add an option to use logits as `hidden_states` in reward modeling.
* Otherwise, if my understanding is correct, this part of the documentation should be removed.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by DarkLight1337** [comment]
@Went-Liang since you originally implemented this, can you give some thoughts?

**Comment by ryokamoi** [comment]
Hi @Went-Liang, any comments would be appreciated!
